pkg/fanal/analyzer/language/python/pip/pip.go

```diff
@@ -41,3 +96,123 @@ func (a pipLibraryAnalyzer) Type() analyzer.Type {
 func (a pipLibraryAnalyzer) Version() int {
 	return version
 }
+
+// licenseList returns list of licenses found in METADATA files in the `site-packages` directory.
+func (a pipLibraryAnalyzer) licenseList() (map[string][]string, error) {
+	var licenses = make(map[string][]string)
+
+	spDir, err := pythonSitePackagesDir()
+	if err != nil {
+		return nil, xerrors.Errorf("Unable to find python `site-packages` directory: %w", err)
+	}
+	requiredMetadata := func(filePath string, _ fs.DirEntry) bool {
+		return strings.HasSuffix(filepath.Dir(filePath), ".dist-info") && filepath.Base(filePath) == "METADATA"
+	}
+
+	// Detect licenses from `site-packages` directory
+	if err = fsutils.WalkDir(os.DirFS(spDir), ".", requiredMetadata, func(path string, d fs.DirEntry, r io.Reader) error {
+		rs, err := xio.NewReadSeekerAt(r)
+		if err != nil {
+			return xerrors.Errorf("Unable to convert reader: %w", err)
+		}
+
+		metadataPkg, _, err := a.metadataParser.Parse(rs)
+		if err != nil {
+			return xerrors.Errorf("metadata parse error: %w", err)
+		}
+
+		// METADATA file contains info about only 1 package
+		licenses[packageID(metadataPkg[0].Name, metadataPkg[0].Version)] = metadataPkg[0].Licenses
+		return nil
+	}); err != nil {
+		return nil, xerrors.Errorf("walk site-packages dir error: %w", err)
+	}
+	return licenses, nil
+}
+
+// pythonSitePackagesDir returns path to site-packages dir
+func pythonSitePackagesDir() (string, error) {
```

Comment:
I'm wondering if Python will fall back to a next priority directory, like $PATH. When running "docker", a shell tries to find "docker" in /bin, /usr/bin, /usr/local/bin, and so on until "docker" is found.
If a package is not found in site-directory, Python may try another site-directory. I'm just guessing. IIUC, the current logic tries one directory found first.

Reply:
IIUC if `site-directory` doesn't contain package - Python just download it.

I saw the follow:
Python 3.11 has been installed on my PC (with `Flask==2.0.0` package in `lib/python3.11/site-packages`).
When i updated to 3.12 and tried to install `Flask==2.0.0` package - Python downloaded this package.

Reply:
Python 3.11 and 3.12 are different versions. It makes sense to download a package again. If you install two packages in the following directories respectively, does Python find both packages?

- `lib/python3.11/site-packages`
- `lib/site-packages` 

Let's say there are `lib/python3.11/site-packages/pkgA` and `lib/site-packages/pkgB`. Can Python 3.11 recognize those packages? Or pkgA only?

Reply:
if `lib/site-packages` contains package - Python still downloads package:

```bash
root@d0f3eacd9026:/# ls -hl /usr/local/lib/site-packages/ | grep flask-3
drwxr-xr-x 2 502 dialout 4.0K May 28 08:17 flask-3.0.3.dist-info

root@d0f3eacd9026:/# pip3 install flask
Collecting flask
  Downloading flask-3.0.3-py3-none-any.whl.metadata (3.2 kB)
Collecting Werkzeug>=3.0.0 (from flask)
...
```

Reply:
OK, thanks for confirming. It's okay, then.